src/main/java/school/faang/user_service/service/RecommendationService.java

```diff
@@ -0,0 +1,121 @@
+            skillOfferRepository.create(skillId, recommendation.getId());
+        }
+
+        return recommendation;
+    }
+
+    public void delete(Long recommendationId) {
+        recommendationRepository.deleteById(recommendationId);
+    }
+
+    public List<RecommendationDto> getAllUserRecommendations(long receiverId) {
+        Page<Recommendation> allByReceiverId = recommendationRepository
+                .findAllByReceiverId(receiverId, Pageable.unpaged());
+        List<Recommendation> recommendations = allByReceiverId.getContent();
+
+        return recommendationMapper.mapToDtoList(recommendations);
+    }
+
+    public List<RecommendationDto> getAllGivenRecommendations(long authorId) {
+        Page<Recommendation> allByAuthorId = recommendationRepository
+                .findAllByAuthorId(authorId, Pageable.unpaged());
+        List<Recommendation> recommendations = allByAuthorId.getContent();
+
+        return recommendationMapper.mapToDtoList(recommendations);
+    }
+
+    private void validateRecommendationForPeriod(RecommendationDto recommendation) {
+        Optional<LocalDateTime> lastRecommendationDate;
+        lastRecommendationDate = getLastRecommendationDate(recommendation.getAuthorId(), recommendation.getReceiverId());
+        lastRecommendationDate.ifPresent(date -> {
+            if (date.plusMonths(PERIOD_TO_ADD_NEW_RECOMMENDATION).isAfter(LocalDateTime.now())) {
+                throw new BusinessException(
+                        String.format("Новая рекомендация может быть дана не ранее, чем через %s месяцев",
+                                PERIOD_TO_ADD_NEW_RECOMMENDATION));
+            }
+        });
+    }
+
+    private void validateSkillsInSystem(RecommendationDto recommendation) {
+        List<Long> skillIds = getSkillIds(recommendation);
+        List<Skill> skillsFromDb = skillRepository.findAllById(skillIds);
+
+        if (skillsFromDb.size() != skillIds.size()) {
+            throw new DataValidationException("Вы предлагаете навыки, которых нет в системе");
+        }
+    }
+
+    private void saveSkillOffers(Long recommendationId, List<Long> skillIds) {
+        for (Long skillId : skillIds) {
+            skillOfferRepository.create(skillId, recommendationId);
+        }
+    }
+
+    private List<Long> getSkillIds(RecommendationDto recommendation) {
+        return recommendation.getSkillOffers().stream()
+                .map(SkillOfferDto::getSkillId)
+                .toList();
+    }
+
+    private Optional<LocalDateTime> getLastRecommendationDate(Long authorId, Long receiverId) {
+        Optional<Recommendation> lastRecommendation = recommendationRepository
+                .findFirstByAuthorIdAndReceiverIdOrderByCreatedAtDesc(authorId, receiverId);
+        if (lastRecommendation.isPresent()) {
+            LocalDateTime date = lastRecommendation.get().getCreatedAt();
+            return Optional.ofNullable(date);
+        } else {
+            return Optional.empty();
+        }
```

Comment:
```suggestion
return lastRecommendation.map(Recommendation::getCreatedAt);
```
Можно одной строчкой заменить)